backend/packages/Upgrade/src/api/services/FeatureFlagService.ts

```diff
@@ -333,4 +334,42 @@ export class FeatureFlagService {
     const includedFeatureFlags = featureFlags.filter(({ id }) => includedFeatureFlagIds.includes(id));
     return includedFeatureFlags;
   }
+
+  public async validateImportFeatureFlags(featureFlagFiles: any, logger: UpgradeLogger): Promise<ValidatedFeatureFlagsError[]> {
+    logger.info({ message: 'Validate feature flags' });
+    const validationErrors = await Promise.allSettled(
+      featureFlagFiles.map(async (featureFlagFile) => {
+        let featureFlag = JSON.parse(featureFlagFile.fileContent);
```

Comment:
We should return or throw an error if this JSON.parse fails. Maybe add it to the incompatibility cases?

Reply:
code updated